### ISSUE TITLE
Transparency and lights

### ISSUE THREAD

**Issue by e2002e** [armory3d/armory#2520]
I have a several transparent shaders that will not compile when using any other light type than 'sun'.
The error is:
/compiled/Shaders/Flame_translucent.frag.glsl:162: 'sampleLight' : no matching overloaded function found 

Voxel shadows is enabled, and it does work neither with it disabled nor with shadow map atlas (these are not compatible).
The reason I want to add another light is because I have emissive materials and they don't light the scene.

Also this should be another issue, but since this regards also transparent materials I'm posting here:
When using volume light and having a transparent material that is untouched by light but a shaft goes in front of it. The material stands out:

![Capture d’écran de 2022-07-18 12-27-53](https://user-images.githubusercontent.com/29781855/179493677-b05297b3-4d8d-4526-b184-a89ba3307ed0.png)

Forward renderer does manage this issue but then I cannot find the correct settings for the glass.

**Comment by MoritzBrueckner** [comment]
Hi, could you please share the blend file or at least the generated `/compiled/Shaders/Flame_translucent.frag.glsl`? What and how many lights do you have in the scene?

**Comment by e2002e** [comment]
[church.zip](https://github.com/armory3d/armory/files/9137582/church.zip)
I packed the textures it should display.
By default it illustrates the issue (volume light and deferred rendering)

**Comment by e2002e** [comment]
I'm sending you the current version of the church,
https://ufile.io/dljk3ove
I had an issue previously with tesselation and had to uncheck 'precompute tangents'. This one's fine though.

**Comment by MoritzBrueckner** [comment]
Does the original error message from your first post

```
/compiled/Shaders/Flame_translucent.frag.glsl:162: 'sampleLight' : no matching overloaded function found
```

still happen for you in that exact file you've shared? Then I'm afraid I can't reproduce it. The volumetric lighting issue should be fixed in https://github.com/armory3d/armory/pull/2523, apart from that I can only reproduce an issue with the tesselation in your file that I will try to fix once I get more time (it's a very weird error message I've never seen before in a similar way).

**Comment by e2002e** [comment]
Yes, same error with a point lamp added. Can you add one ?
Really good work with the fix. I'm going to try this today.

**Comment by MoritzBrueckner** [comment]
Thanks, now I can confirm it.  I will investigate :)

**Comment by e2002e** [comment]
To pinpoint the issue, material set to an oppacity of less than one will yield this error, whatever blend mode is used.

**Comment by e2002e** [comment]
Made a fix in #2531, ssrs seems ok, i am not sure for microshadowing

**Comment by e2002e** [comment]
This is still an issue, gbuffers are not set correctly for the shader pass,
I am looking for the function that calls drawShaders(context) in iron.RenderPath, with something else that the post process'.
All those shaders in the compiled/Shaders folder are being treated at some point but I can't find where.
I think the add_uniforms concerning the gbuffers are uneffective and need the 'link=' parameter, though I'm unable to find the latters.

**Comment by MoritzBrueckner** [comment]
Hi, sorry, I'm quite busy right now, so I don't have much time for the next few days to look into this issue (also regarding your comments in the related PR).

> I am looking for the function that calls drawShaders(context) in iron.RenderPath, with something else that the post process'. All those shaders in the compiled/Shaders folder are being treated at some point but I can't find where.

Can you elaborate? An individual mesh is rendered in `iron.object.MeshObject.render()`.

As for the uniforms I'm not entirely sure when it's sufficient to only include them in the shader's .json file, but for accessing the depth buffer you may need something similar to https://github.com/armory3d/armory/pull/2474, but without the sorting since translucent materials are drawn after all other meshes anyway.

**Comment by e2002e** [comment]
I figured it out, and set the buffers int Inc.drawTranslucency before the mesh is drawn, result is that microshadowing are taken into account on transparent objects, but still ignored when used in parallel with ssrs.
